app/controllers/submissions_controller.rb

```diff
@@ -481,41 +483,31 @@ def download
   end
 
   ##
-  # Download all files from groupings with id in +params[:groupings]+ in a .zip file.
+  # Prepare all files from groupings with id in +params[:groupings]+ to be downloaded in a .zip file.
   ##
-  def download_groupings_files
+  def zip_groupings_files
     assignment = Assignment.find(params[:assignment_id])
 
-    ## create the zip name with the user name to have less chance to delete
-    ## a currently downloading file
-    short_id = assignment.short_identifier
-    zip_name = Pathname.new(short_id + '_' + current_user.user_name + '.zip')
-    zip_path = Pathname.new('tmp') + zip_name
-
-    ## delete the old file if it exists
-    File.delete(zip_path) if File.exist?(zip_path)
-
     groupings = Grouping.includes(:group, :current_submission_used).where(id: params[:groupings]&.map(&:to_i))
```

Comment:
This is kind of awkward now that we're just passing in grouping ids to the job.

1. You can definitely remove the `.includes` here and move it into the job itself.
2. You can `pluck(:id)`.
3. Since it seems this query would just be acting as a filter (e.g. removing invalid ids), I think it would be nice to do `assignment.groupings.where(...)` rather than `Grouping.where(...)`. In general I'm trying to make a change to do more association querying for these nested routes.

(Also, this comment affects the update on line 496 too.)

Reply:
for 2. I'm going to stick with `.ids` instead of `pluck(:id)` since it does the exact same thing and I like the more compact syntax if that's ok with you:

```ruby
irb(main):005:0> a.groupings.pluck(:id)
   (1.1ms)  SELECT "groupings"."id" FROM "groupings" WHERE "groupings"."assessment_id" = $1  [["assessment_id", 5]]
=> [463, 464, 465, 466, 467, 468, 469, 470, 471, 472, 473, 474, 475, 476, 477]
irb(main):006:0> a.groupings.ids
   (1.0ms)  SELECT "groupings"."id" FROM "groupings" WHERE "groupings"."assessment_id" = $1  [["assessment_id", 5]]
=> [463, 464, 465, 466, 467, 468, 469, 470, 471, 472, 473, 474, 475, 476, 477]
```